### PR TITLE
Added shotgun pooling functions

### PR DESCRIPTION
**Pull request by tanaes**
Adds functions and tests for calculating qPCR concentrations and pooling libraries.

### CODE REVIEW

File: knimin/lib/shotgun.py [hunk 1]
```diff
@@ -0,0 +1,160 @@
...
+        The total volume to pool (uL)
+
+    Returns
+    -------
+    np.array of floats
+        A 2D array of floats
+    """
+    per_sample_vol = (total_vol / sample_concs.size) * 1000.0
+    
+    sample_vols = np.zeros([sample_concs.shape[0], sample_concs.shape[1]]) +
```
Comment (wasade) [inline knimin/lib/shotgun.py]: `sample_vols = np.zeros_like(sample_concs) + per_sample_vol`

Comment (wasade) [inline knimin/lib/shotgun.py]: ...non blocking request

File: knimin/lib/shotgun.py [hunk 2]
```diff
@@ -0,0 +1,160 @@
...
+    -------
+    sample_vols: np.array of floats
+        the volumes in nL per each sample pooled
+    """
+
+    if sample_fracs is None:
+        sample_fracs = np.ones(sample_concs.shape) / sample_concs.size
+    
+    # get samples above threshold
+    sample_fracs_pass = sample_fracs.copy()
```
Comment (wasade) [inline knimin/lib/shotgun.py]: this will throw an exception if `sample_fracs is None`. In addition to adding logic to resolve this, can a test be added which explicitly passes in `sample_fracs=None` to ensure this branch is exercised?

Comment (wasade) [inline knimin/lib/shotgun.py]: Nope, I looked at this wrong.

File: knimin/lib/shotgun.py (hunk 1 above)
Comment (tanaes) [inline knimin/lib/shotgun.py]: Fixed!

File: knimin/lib/shotgun.py (hunk 2 above)
Comment (tanaes) [inline knimin/lib/shotgun.py]: Fixed, although still don't have tests to exercise additional cases here (passing / not passing non-default values)

**Comment by wasade** [comment]
I'm :+1:

**Comment by tanaes** [comment]
@josenavas These tests are passing now, and should be ready to merge.
